### DISCUSSION TITLE
Should we have a way to dynamically passing Input name ?

### DISCUSSION THREAD

**Discussion by dnakhoa** [NillionNetwork/discussions#13] [Ideas]
A input name on NADA lang is a static value, so whenever we create a NADA program we need to specified the name. Ex: {"A" : 10}.  Instead of having to hardcode "A" as an Input name in our NADA program, like:

` A = SecretInteger(Input(name="A", party=party1)) 
` 
 we can pass "A" to an input and use that as a variable to be use as Input name. 
 Ex:
` A = SecretInteger(Input(name=p_name, party=party1))`

**Comment by wwwehr** [comment]
as I understand it you are wanting to have dynamic naming in your program.

lets think of these named program inputs as "named" parameters, like in python. we don't have a way of altering these during a circuit execution BUT I think maybe you could design your system so that it actually

1. creates the program code dynamically using a template (like jinja2)
2. compile the program w/ pynadac
3. upload the program (using py client or nilcli)
4. use the program in your app

consider the above as if you had a webservice managing signals from the frontend and injecting/uploading to the network, getting the assigned id and passing back to ui app...

![mermaid-diagram-2024-03-05-152623](https://github.com/nillion-oss/builder-resources/assets/33910651/5206f63d-7ad7-4ac9-8327-5ce9dd53bfb8)


hope this helps!
